src/kwp2000/start_diagnostic_session.rs

```diff
@@ -37,30 +36,37 @@ pub enum SessionType {
     /// Extended diagnostics mode. Every service is available here
     ExtendedDiagnostics,
     /// Custom diagnostic mode not in the KWP2000 specification
-    Custom(u8),
+    Custom { id: u8 },
```

Comment:
For consistency, you may want to use the same ByteWrapper enum from auto-uds. This way we can factor-out the KWP or any other enums into separate libs too

Reply:
Making the wrapper its own crate seems silly - it's a tiny enum. I think a better path would be to set up another repo similar to auto_uds, e.g. auto_kwp (or is it auto_kwp2000?), and just copy/paste the utils file there. Having two implementations of utils in two separate crates is much better than to maintain a common dependency when its such a tiny code that doesn't require significant testing.  What's more important is to have approach consistency - i.e. all constants defined/parsed/used in the same way/style, as that would simplify any kind of reasoning about the code.